classes/Commands/RestoreCommand.php

```diff
@@ -80,9 +84,59 @@ protected function execute(InputInterface $input, OutputInterface $output): ?int
 
             return $exitCode;
         } catch (Exception $e) {
-            $this->logger->error('An error occurred during the restoration process: ' . $e->getMessage());
+            $this->logger->error('An error occurred during the restoration process');
+            throw $e;
+        }
+    }
+
+    /**
+     * @throws Exception
+     */
+    private function selectBackupInteractive(InputInterface $input, OutputInterface $output): string
+    {
+        $backups = $this->getBackups();
 
-            return ExitCode::FAIL;
+        if (empty($backups)) {
+            throw new BackupException('No store backup files found in your dedicated directory');
         }
+
+        $formattedBackups = array_map(function ($backupName) {
+            return $this->parseBackupMetadata($backupName);
+        }, $backups);
+
+        // Most recent first
+        usort($formattedBackups, function ($a, $b) {
+            return $b['timestamp'] <=> $a['timestamp'];
+        });
```

Comment:
I think this part will be useful for the web as well, if you can extract this too.